app/src/main/res/navigation/nav_graph.xml

```diff
@@ -172,7 +172,11 @@
         android:id="@+id/googleSignInWidget"
         android:name="com.github.HumanLearning2021.HumanLearningApp.view.GoogleSignInWidget"
         android:label="@string/nav_graph_label_login"
-        tools:layout="@layout/fragment_google_sign_in" />
+        tools:layout="@layout/fragment_google_sign_in" >
+        <action
+            android:id="@+id/action_googleSignInWidget_to_homeFragment"
+            app:destination="@id/homeFragment" />
```

Comment:
```suggestion
            app:destination="@id/homeFragment"
            app:popUpTo="@+id/homeFragment"
            app:popUpToInclusive="true"
            />
```

Reply:
When I press back it takes me back to login fragment. I think this should fix it.